pkg/endpoint/policy.go

```diff
@@ -879,9 +884,68 @@ func (e *Endpoint) runIdentityToK8sPodSync() {
 	)
 }
 
+// This synchronizes the key-value store with a mapping of the endpoint's IP
+// with the numerical ID representing its security identity.
+func (e *Endpoint) runIPIdentitySync(endpointIP addressing.CiliumIP) {
+
+	if endpointIP == nil {
+		return
+	}
+
+	var addressFamily string
+	if endpointIP.IsIPv6() {
+		addressFamily = "ipv6"
+	} else {
+		addressFamily = "ipv4"
+	}
+
+	e.controllers.UpdateController(fmt.Sprintf("sync-%s-identity-mapping (%d)", addressFamily, e.ID),
+		controller.ControllerParams{
+			DoFunc: func() error {
+
+				ipKey := path.Join(ipcache.IPIdentitiesPath, ipcache.AddressSpace, endpointIP.String())
+				e.Mutex.RLock()
```

Comment:
We should check `if e.state == StateDisconnected || e.state == StateDisconnecting{` and return to account for the case where the endpoint has already been deleted again but the controller was already scheduled to run

Reply:
Done.